### ISSUE TITLE
ChangeParentPom: Maven authentication issues

### ISSUE THREAD

**Issue by eugene-sadovsky** [openrewrite/rewrite#2527]
Hello, 
I'm having an issue with ChangeParentPom recipe. My pom file is hosted in artifactory which requires authentication. When I apply the recipe I get this in project pom.xml and the version is not getting upgraded

```
    <!--~~(my.org:my-parent-pom failed. Unable to download metadata. Tried repositories:
https://artifactory.some.host/artifactory/libs-release/: HTTP 401
https://artifactory.some.host/artifactory/libs-snapshot/: HTTP 401
https://artifactory.some.host/artifactory/qa-release-local/: HTTP 401
https://artifactory.some.host/artifactory/libs-release-local/: HTTP 401
https://artifactory.some.host/artifactory/libs-snapshot-local: HTTP 401
https://repo.maven.apache.org/maven2: HTTP 404)~~>--><parent>
        <groupId>my.org</groupId>
        <artifactId>my-parent-pom</artifactId>
        <version>2.11.0</version>
    </parent>
```

My ~/.m2/settings.xml has required auth and all works fine when building the project.

I've traced the issue [to here](https://github.com/openrewrite/rewrite/blob/b25cfc293bc95830dc23cd9820bf97bf38fbfea9/rewrite-maven/src/main/java/org/openrewrite/maven/internal/MavenPomDownloader.java#L745)

passed `MavenRepository repository` already has credentials but then this methods nulls them. I've modified method as follows, which solves my issue
```
private MavenRepository applyAuthenticationToRepository(MavenRepository repository) {
    if (hasCredentials(repository)){
        return repository;
    }
    return MavenRepositoryCredentials.apply(ctx.getCredentials(mavenSettings), repository);
}
```

Not entirely sure whether this is a bug or I am missing something in my config

**Comment by tkvangorder** [comment]
Hi @eugene-sadovsky 

I think this is a duplicate of #2515 . Can you please try to use version `4.38.1` of the rewrite-maven-plugin?

**Comment by eugene-sadovsky** [comment]
Hello Tyler, 

Thank you for a quick response. Yes, I can confirm this is a duplicate and has been resolved in `4.38.1` version 👍🏼

**Comment by tkvangorder** [comment]
Excellent!
